x-pack/plugins/maps/public/shared/layers/styles/components/heatmap/legend/heatmap_legend.js

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import { i18n } from '@kbn/i18n';
+import { ColorGradient } from '../../color_gradient';
+import { StyleLegendRow } from '../../style_legend_row';
+import {
+  DEFAULT_RGB_HEATMAP_COLOR_RAMP,
+  DEFAULT_HEATMAP_COLOR_RAMP_NAME,
+  HEATMAP_COLOR_RAMP_LABEL
+} from '../heatmap_constants';
+
+export function HeatmapLegend({ colorRampName, label }) {
+  const header = colorRampName === DEFAULT_HEATMAP_COLOR_RAMP_NAME
+    ? <ColorGradient colorRamp={DEFAULT_RGB_HEATMAP_COLOR_RAMP}/>
+    : <ColorGradient colorRampName={colorRampName}/>;
+
+  return (
+    <StyleLegendRow
+      header={header}
+      minLabel={
+        i18n.translate('xpack.maps.heatmapLegend.coldLabel', {
+          defaultMessage: 'cold'
```

Comment:
Because the heatmap combines values into hot spots. So the values do not range from min to max. The values range to clumping together the maxes. Heatmap are not about absolute values. They are more concerned with coverage. The text is just there to let users know which color signals more data and which color signals less data